plugin/dnstap/io.go

```diff
@@ -103,9 +101,7 @@ func (d *dio) close() { close(d.quit) }
 func (d *dio) write(payload *tap.Dnstap) error {
 	if d.enc == nil {
 		atomic.AddUint32(&d.dropped, 1)
-		return nil
-	}
-	if err := d.enc.writeMsg(payload); err != nil {
+	} else if err := d.enc.writeMsg(payload); err != nil {
```

Comment:
This change does not appear to present any obvious optimization.

I prefer the early return and separate if instead of else-if.